src/db/migrate.rs

```diff
@@ -322,6 +323,32 @@ fn migrate_inner(
         ),
     ];
 
+    if let ApplyMode::Permanent = apply_mode {
+        migrations.push(migration!(
+            context,
+            // version
+            10,
+            // description
+            "",
+            // upgrade query
+            "
+                CREATE FUNCTION normalize_crate_name(VARCHAR)
+                RETURNS VARCHAR
+                AS $$
+                    SELECT LOWER(REPLACE($1, '_', '-'));
+                $$ LANGUAGE SQL;
+
+                CREATE UNIQUE INDEX crates_normalized_name_idx
+                    ON crates (normalize_crate_name(name));
+            ",
+            // downgrade query
+            "
+                DROP FUNCTION normalize_crate_name;
+                DROP INDEX crates_normalized_name_idx;
+            "
```

Comment:
I think the order of the downgrade query will have to be switched or DROP FUNCTION will give an error that the index depends on it.